### ISSUE TITLE
Renovate should not use pre-release pre-beta versions of container images

### ISSUE THREAD

**Issue by MarkSymsCtx** [renovatebot/renovate#17325]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of the platform you run.

Bitbucket Server

### Was this something which used to work for you, and then stopped?

I never saw this working

### Describe the bug

Renovate generated a pull request to update the base image for a container from Fedora:35 to Fedora:38. Fedora 38 is not released and isn't due to be released until 2023-04-18. Even Fedora 37 is not yet released, due on 2022-10-18 so the latest release that should be under consideration is Fedora 36 and the current Fedora 35 image is whilst not the latest release still under support.

```
This PR contains the following updates:</div>

Package | Type | Update | Change
-- | -- | -- | --
<redacted internal server>/fedora/fedora | final | major | 35 -> 38
```

I don't have the version of renovate being used as this is hosted by a centralised team.

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

### Have you created a minimal reproduction repository?

I have linked to a minimal reproduction repository in the bug description

**Comment by Gabriel-Ladzaretti** [comment]
Based on you bug description I'm assuming we are talking about the docker datasource. if so, this might not be a bug as the versioning docker uses is "a loose semver" type. therefore it is not aware of any scheduling.

similar discussion you can find here
 - https://github.com/renovatebot/renovate/discussions/17082
 
and this [comment](https://github.com/renovatebot/renovate/discussions/17082#discussioncomment-3376011)

**Comment by MarkSymsCtx** [comment]
> Based on your bug description I'm assuming we are talking about the docker datasource. if so, this might not be a bug as the versioning docker uses is "a loose semver" of type. therefore it is not aware of any scheduling.
> 
> similar discussion you can find here
> 
> * [LTS awareness #17082](https://github.com/renovatebot/renovate/discussions/17082)
> 
> and this [comment](https://github.com/renovatebot/renovate/discussions/17082#discussioncomment-3376011)
> 
> EDIT: couldn't find no ver 38 on docker hub, please provide logs/repro for further investigation

Yes, precisely but that just makes Renovate not fit for purpose.

**Comment by Gabriel-Ladzaretti** [comment]
You can try restricting it with `packageRules` and `allowedversions`
https://docs.renovatebot.com/configuration-options/#allowedversions

**Comment by nabeelsaabna** [comment]
also: https://github.com/renovatebot/renovate/issues/5773
